scripts/update-cards.js

```diff
@@ -1,29 +1,153 @@
+    delete typeMapClone.ALL;
+    const validTypes = Object.values(typeMapClone);
+
+    const validKeywords = [`(?<!^(\\[x\\]<b>|<b>))${keywordsMap.TAUNT}`]; // Don't count a taunt synergy if it has taunt.
+
+    const validSynergies = validTypes.concat(validKeywords);
+    const synergyRegex = new RegExp(`\\b(?:${validSynergies.join('|')})`, 'g');
+    const getSynergies = (card) => {
+        let synergies = [];
+        const type = typeMap[card["race"]];
+        const text = card["text"];
+
+        // Only type-less minions count for synergy
+        if (text && !type) {
+            synergies = text.match(synergyRegex) || [];
+
+            // Check for Lightfang
+            const isLightFang = text.match(/each minion type/gi);
+            if (isLightFang) {
+                synergies.push(typeMap["ALL"]);
+            }
+
+            // Filter out duplicates.
+            synergies = [...new Set(synergies)];
+        }
+
+        return synergies;
+    };
+
+    let formattedMinions = cardsArray.map((card) => {
+        const type = typeMap[card["race"]];
+        const keywords = getKeywords(card);
+        const synergies = getSynergies(card);
+        // Filter out duplicates between keywords and synergies.
+        const keywordsAndSynergies = synergies.length ? synergies.concat(keywords) : keywords;
+        const combined = [...new Set(keywordsAndSynergies)];
+        if (type) {
+            combined.unshift(type);
+        }
+
+        const formattedMinion = {
+            "Name": card["name"],
+            "Tier": card["techLevel"].toString(),
+            "Attack": card["attack"].toString(),
+            "Health": card["health"].toString(),
+            "Cost": card["cost"].toString(),
+            "Legendary": card["elite"] && "Y",
+            "Type": type,
+            "Keywords": keywords.length ? keywords.join(', ') : undefined,
+            "Synergy": synergies.length ? synergies.join(', ') : undefined,
+            "Combined": combined.length ? combined.join(', ') : undefined,
+            "ID": card["id"]
+        };
+
+        // Remove keys with undefined values
+        Object.keys(formattedMinion).forEach(key => formattedMinion[key] === undefined && delete formattedMinion[key]);
+
+        return formattedMinion;
+    });
+
+    fs.writeFile(
+        "./src/minions.js",
+        `const minions = ${JSON.stringify(formattedMinions, null, 4)}\n\nexport default minions;`,
+        (err) => err ? console.log(err) : console.log("The minions.js file has been updated. 🙌 🎉 🙌 🎉")
+    );
+};
+
 const updateCardsJsonFile = () => {
     const cardsArray = JSON.parse(cardsRequest.responseText);
     
     // Filter for only BG minions
     const bgCardsArray = []
     cardsArray.forEach((card) => {
-        if(card["battlegroundsPremiumDbfId"]) {
-            bgCardsArray.push({ id: card["id"], name: card["name"] })
+        // Retired BG minions don't have a techLevel anymore
+        if(card["techLevel"]) {
```

Comment:
filtering for minions with techLevel solves for excluding retired/removed BG cards